core/src/main/java/co/cask/wrangler/steps/transformation/functions/DDL.java

```diff
@@ -0,0 +1,406 @@
+   */
+  public static Schema parse(StructuredRecord record) {
+    return record.getSchema();
+  }
+
+  /**
+   * Converts a {@link StructuredRecord} to JSON string.
+   *
+   * @param record to be converted to JSON
+   * @return JSON string representation of structured record.
+   * @throws IOException thrown if there are any issues with parsing.
+   */
+  public static JsonElement toJson(StructuredRecord record) throws IOException {
+    return JSON.parse(StructuredRecordJsonConverter.toJsonString(record));
+  }
+
+  /**
+   * Given a {@link StructuredRecord} and the name of the field, it checks, if the
+   * field exists in the {@link StructuredRecord}.
+   *
+   * @param record in which the field needs to be checked.
+   * @param name of the field to be checked.
+   * @return true if present and it's a record, else false.
+   */
+  public static boolean hasField(StructuredRecord record, String name) {
+    Schema.Field field = record.getSchema().getField(name);
+    if (field == null) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Drops a path from the schema to generate a new {@link Schema}.
+   *
+   * @param schema to be modified.
+   * @param path to be removed from {@link Schema}.
+   * @return modified {@link Schema}
+   */
+  public static Schema drop(Schema schema, String path) {
+    return drop(schema, path, path);
+  }
+
+  public static Schema drop(Schema schema, String path, String ... paths) {
+    Schema mutatedSchema = drop(schema, path, path);
+    for (String otherPath : paths) {
+      mutatedSchema = drop(mutatedSchema, otherPath, otherPath);
+    }
+    return mutatedSchema;
+  }
+
+  private static Schema drop(Schema schema, @Nullable String path, String fullPath) {
+    if (path == null) {
+      return schema;
+    }
+
+    int dotIndex = path.indexOf('.');
+    String recordField = dotIndex > 0 ? path.substring(0, dotIndex) : path;
+    String nextPath = dotIndex > 0 ? path.substring(dotIndex + 1) : null;
+
+    int arrayIndex = -1;
+    int bracketIndex = recordField.indexOf('[');
+    if (bracketIndex > 0) {
+      if (!recordField.endsWith("]")) {
+        throw new IllegalArgumentException(
+          String.format("Invalid field '%s' in path '%s'. An array index start with '[' and must end with a ']'",
+                        recordField, fullPath));
+      }
+      String indexStr = recordField.substring(bracketIndex + 1, recordField.length() - 1);
+      try {
+        arrayIndex = Integer.parseInt(indexStr);
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException(
+          String.format("Invalid array index '%s' for field '%s' in path '%s'. Must be a valid integer.",
+                        indexStr, recordField, fullPath));
+      }
+      recordField = recordField.substring(0, bracketIndex);
+    }
+
+    List<Schema.Field> fields = new ArrayList<>();
+    for (Schema.Field field : schema.getFields()) {
```

Comment:
The `schema` may not be of record type. Calling `getFields()` on non-record type will have `null` value returned.